Sources/Datadog/RUM/RUMVitals/VitalRefreshRateReader.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+/// A class reading the refresh rate (frames per second) of the main screen
+internal class VitalRefreshRateReader {
+    private var observers = [VitalObserver]()
+    private var displayLink: CADisplayLink?
+    private var lastFrameTimestamp: CFTimeInterval?
+    private(set) var isRunning = false
+
+    init(notificationCenter: NotificationCenter = .default) {
+        notificationCenter.addObserver(
+            self,
+            selector: #selector(appWillResignActive),
+            name: UIApplication.willResignActiveNotification,
+            object: nil
+        )
+        notificationCenter.addObserver(
+            self,
+            selector: #selector(appDidBecomeActive),
+            name: UIApplication.didBecomeActiveNotification,
+            object: nil
+        )
+    }
+
+    deinit {
+        stop()
+    }
+
+    /// `VitalRefreshRateReader` keeps pushing data to its `observers` at every new frame.
+    /// - Parameter observer: receiver of refresh rate per frame.
+    func register(_ observer: VitalObserver) {
+        DispatchQueue.main.async {
+            self.observers.append(observer)
+        }
+    }
+
+    /// `VitalRefreshRateReader` stops pushing data to `observer` once unregistered.
+    /// - Parameter observer: already added observer; otherwise nothing happens.
+    func unregister(_ observer: VitalObserver) {
+        DispatchQueue.main.async {
+            self.observers.removeAll { existingObserver in
+                return existingObserver === observer
+            }
+        }
+    }
+
+    /// Starts listening to frame paints.
+    /// - Throws: only if `UIScreen.main` cannot generate its `CADisplayLink`
+    func start() throws {
```

Comment:
> IDK who will be the caller of that method
i add `start()` to `init()` so that `Reader` will be always running ✅ 

> this method should be kept simple and marked non-throwing
`UIScreen.main.displayLink()` throws but `CADisplayLink()` constructor doesn't throw.
i changed to non-throwing method ✅